lib/driver_sht41/sht41.c

```diff
@@ -24,69 +21,65 @@ static int64_t sht41_start_read_req(__unused alarm_id_t id, void *user_data);
-
-static void sht41_on_error(const sht41_error_code error_type, uint32_t i2c_error_code);
+static void sht41_on_error(const sht41_error_code error_type);
 
 // SHT41 data fields
 static const uint8_t write_cmd[] = { SHT41_WRITE_CMD };
 static uint8_t rx_buf[SHT41_READ_LENGTH];
 
 // requests
-static struct async_i2c_request wr_req = {
-    .i2c_num = BOARD_I2C,
-    .address = SHT41_I2C_ADDR,
-    .nostop = false,
-    .tx_buffer = write_cmd,
-    .rx_buffer = NULL,
-    .bytes_to_send = 1,
-    .bytes_to_receive = 0,
-    .completed_callback = &sht41_on_write,
-    .failed_callback = &on_async_i2c_error,
-    .next_req_on_success = NULL,
-    .user_data = NULL  // TODO double check one this one
-};
-
-static struct async_i2c_request rd_req = {
-    .i2c_num = BOARD_I2C,
-    .address = SHT41_I2C_ADDR,
-    .nostop = false,
-    .tx_buffer = NULL,
-    .rx_buffer = rx_buf,
-    .bytes_to_send = 0,
-    .bytes_to_receive = 6,
-    .completed_callback = &sht41_on_read,
-    .failed_callback = &on_async_i2c_error,
-    .next_req_on_success = NULL,
-    .user_data = NULL  // FIXME double check on this one
-};
+// static struct async_i2c_request wr_req = { .i2c_num = BOARD_I2C,
+//                                            .address = SHT41_I2C_ADDR,
+//                                            .nostop = false,
+//                                            .tx_buffer = write_cmd,
+//                                            .rx_buffer = NULL,
+//                                            .bytes_to_send = 1,
+//                                            .bytes_to_receive = 0,
+//                                            .completed_callback = &sht41_on_write,
+//                                            .failed_callback = &on_async_i2c_error,
+//                                            .next_req_on_success = NULL,
+//                                            .user_data = NULL };
+
+// static struct async_i2c_request rd_req = { .i2c_num = BOARD_I2C,
+//                                            .address = SHT41_I2C_ADDR,
+//                                            .nostop = false,
+//                                            .tx_buffer = NULL,
+//                                            .rx_buffer = rx_buf,
+//                                            .bytes_to_send = 0,
+//                                            .bytes_to_receive = 6,
+//                                            .completed_callback = &sht41_on_read,
+//                                            .failed_callback = &on_async_i2c_error,
+//                                            .next_req_on_success = NULL,
+//                                            .user_data = NULL };
 
 static struct sht41_data {
-    int16_t temp;
-    int16_t rh;
-    absolute_time_t reading_expiration;  // TODO nned to init in the init()
+    float temp;
+    float rh;
+    absolute_time_t reading_expiration;
+    struct async_i2c_request read_req;
```

Comment:
You don't need two requests for read and write, you can Instead combine them into one, and change the required fields before submitting new requests